### ISSUE TITLE
Moving blocks on mobile is disorienting

### ISSUE THREAD

**Issue by hedgefield** [WordPress/gutenberg#4758]
## Issue overview
Moving blocks with the arrow buttons on mobile is kind of a disorienting experience right now. The block you're moving stays in the exact same place on the screen, but everything around it changes (because the focus is moved in step with the block itself).

I understand why it works like this - if you keep focus on where you were in the document you could lose sight of the block, and especially when moving it a long way you'd have to keep scrolling and scrolling to keep up. But is there a better way? And, to confuse things, this _does seem to be_ how it works on Android?

## Current behavior
The biggest cause of the disorientation for me is that the translation is instant. Suddenly all kinds of elements change while it seems like the block you were trying to move hasn't responded at all.

## Expected behavior
Is a tiny animation possible? Lifting up the block slightly and scrolling the content below it.

Ideally, this would be a drag-and-drop interaction, but is that possible? I looked at how notion does it, but I couldn't get the drag-and-drop to work in the mobile browser, only in the app.

## Video
https://photos.app.goo.gl/lRNAyMGWKdJdiqyJ2

**Comment by karmatosed** [comment]
> Is a tiny animation possible? Lifting up the block slightly and scrolling the content below it.

I worry about us adding animation here. What else could work over animation that wouldn't add page load and weight?

**Comment by hedgefield** [comment]
> What else could work over animation that wouldn't add page load and weight?

Well.. true drag-n-drop? If all we have is the arrow buttons, the animation would be needed to 'fake' the effect of moving the block. But if we can actually move it by hand, we don't need animation. I've seen [work being done on dragging blocks](https://github.com/WordPress/gutenberg/issues/38), when that gets to a workable state it would be nice to do a user test for it on mobile.
At least I _hope_ it gets to that point, all my go-to places for drag-n-drop on the web (notion, trello, meistertask) don't support that behavior in their webapp...

Other than that, the most basic method I could imagine is to keep the focus on your current place in the document when you tap the arrow buttons on a block, and then quickly scroll the document towards the new location of the block, but I worry what happens when the user interrupts that process.

I think for now let's await what happens with #38 unless someone else has a good idea?

**Comment by karmatosed** [comment]
Shall we close this one then and think about thing after #38?

**Comment by hedgefield** [comment]
Looking more closely at the PR associated with #38 mobile support for drag-n-drop sounds like a 'stretch goal' that needs further investigation. So I'd like to keep this one open as a discussion point at least until the meeting wednesday so we can ask for other ideas, and if none arise, we can close and see how https://github.com/WordPress/gutenberg/pull/4115 shakes out, sound good?

**Comment by karmatosed** [comment]
If drag and drop works on desktop, then it will work on mobile. I don't think it's being seen as a stretch goal. Right now this needs it in desktop before an investigation, otherwise it's a little bit like charting territory we don't know about. That's why I'm suggesting we close this and focus discussion on drag and drop across all devices. A note, we should stop considering it as separate instances. Functionality should just work across all devices.

**Comment by hedgefield** [comment]
> If drag and drop works on desktop, then it will work on mobile.

I don't think we can assume that. Dragging with touch gestures is technically different from dragging with a mouse. Again, demonstrated by the fact that none of the big SCRUM services support dragging on mobile browsers, and #4115 which has desktop drag working but mobile support is listed as a 'non-blocking area of investigation' at the bottom of the list.

I agree with you it _should_ work on both and we should treat is as such, and therefore we need to have the discussion that we shouldn't release drag-and-drop if it doesn't work on mobile, the place where it arguably offers the most benefit. But I didn't want to bloat or delay the PR by having the discussion there, so here we are. But if you can assure that this will be taken care of, then feel free to consider this discussed and close the issue.

**Comment by karmatosed** [comment]
The principle we've taken is to work on all devices, so yes let's follow that into the existing path. Closing for now.
